### DISCUSSION TITLE
Resolving divergent alembic migrations

### DISCUSSION THREAD

**Discussion by bendnorman** [catalyst-cooperative/pudl#3319] [Ideas]
What is our strategy for resolving divergent alembic migrations when we merge `main` into a feature branch? I see three options:
1. Whenever there are divergent heads, remove all migration versions and create a new migration that recreates all metadata.
2. Remove and recreate the migrations created on the feature branch.
3. Merge the migrations using [alembic branch merges](https://alembic.sqlalchemy.org/en/latest/branches.html#merging-branches).

Option three feels like the "right" way to resolve divergent migrations, though it might create some additional complexity when we don't really need a full migration history. Option 1 seems like the simplest option but would require contributors to delete their PUDL database and rerun the `alembic upgrade head.` Maybe that's not a huge issue because it doesn't take that long to repopulate the database. 

I think I prefer option 1 because it is the simplest. Maintaining a full migration history doesn't provide much benefit to us.

**Comment by zaneselvans** [comment]
I also vote for option 1, but I also delete my DB almost every day, sometimes multiple times.  One can also repopulate the outputs by pulling down the nightly build outputs if you're just working on changing a single asset or a small group of assets.
